connect/runtime/src/main/java/org/apache/kafka/connect/runtime/TransformationChain.java

```diff
@@ -37,9 +41,11 @@ public TransformationChain(List<Transformation<R>> transformations, RetryWithTol
     public R apply(R record) {
         if (transformations.isEmpty()) return record;
 
-        for (final Transformation<R> transformation : transformations) {
+        for (Transformation<R> transformation : transformations) {
             final R current = record;
 
+            log.trace("Applying transformation {} to {}",
+                transformation.getClass().getSimpleName(), record);
```

Comment:
I don't think you're going to want to use `getSimpleName()`. I think for a nested class that only returns the nested class name, and we use nested classes as a pattern with transformations to handle keys and values without duplicating a bunch of code, e.g. https://github.com/apache/kafka/blob/trunk/connect/transforms/src/main/java/org/apache/kafka/connect/transforms/ReplaceField.java#L194